services/github/github-go-mod.service.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const { renderVersionBadge } = require('../version')
+const { ConditionalGithubAuthV3Service } = require('./github-auth-service')
+const { fetchRepoContent } = require('./github-common-fetch')
+const { documentation } = require('./github-helpers')
+const { InvalidResponse } = require('..')
+
+const goVersionRegExp = new RegExp('^go (.+)$', 'm')
+
+module.exports = class GithubGoModGoVersion extends ConditionalGithubAuthV3Service {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'github/go-mod/go-version',
+      pattern: ':user/:repo/:branch*',
```

Comment:
We should also include an optional query parameter folks can use to specify the path to the mod file for cases where it's not in the root of the repo/multiple packages/etc.

https://github.com/badges/shields/issues/3960#issuecomment-546084081

Reply:
Coincidentally I just did something really similar for the package.json dependency badge in #4233! Feel free to refer to that to see how to do this.

Reply:
Similar as in #4233 I have added a query param allowing to specify a filename. I guess accepting only a path would suffice though.

Reply:
How would that be different?

I thought about calling this parameter `path` but thought `filename` seemed clearer.

Reply:
Instead of requiring a client to pass a path containing each time `go.mod` part, you could allow him to pass only a path to a directory where `go.mod` resides. Provided the filename `go.mod` never changes, this would mean skipping redundant information as it's known upfront (the error messages already assume the filename is `go.mod`). For example instead of `filename=src/go.mod` you would have `path=src`. What do you think?

Reply:
Ahh, I gotcha. I think it's clearer to include `/go.mod` in the parameter. It makes it self-documenting.

Reply:
Oki doki, let's keep it that way.